hotham-openxr-client/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+// TODO Safety doc would be nice
+#![allow(clippy::missing_safety_doc)]
+
+mod client;
+
+mod space_state;
+mod state;
+
+use std::ffi::c_char;
+
+use crate::client::*;
+
+use openxr_sys::{loader, pfn, Instance, Result};
+
+type DummyFn = unsafe extern "system" fn() -> Result;
+
+pub unsafe extern "system" fn get_instance_proc_addr(
+    _instance: Instance,
+    name: *const c_char,
+    function: *mut Option<pfn::VoidFunction>,
+) -> Result {
+    use std::{ffi::CStr, intrinsics::transmute};
+
+    let name = CStr::from_ptr(name);
+    let name = name.to_bytes();
+    if name == b"xrGetInstanceProcAddr" {
```

Comment:
Could this be a match expression instead?

Reply:
Eh. I've tried a couple different implementations of this - the if statements are messy and yuck, but they work.